src/material/autocomplete/autocomplete-trigger.ts

```diff
@@ -858,6 +861,59 @@ export abstract class _MatAutocompleteTriggerBase
     // but the behvior isn't exactly the same and it ends up breaking some internal tests.
     overlayRef.outsidePointerEvents().subscribe();
   }
+
+  private _trackedModals = new Set<Element>();
+
+  /**
+   * Some browsers won't expose the accessibility node of the listbox overlay if there is an
+   * `aria-modal` and the overlay element is outside of it. This method works around the issue by
+   * pointing the `aria-owns` of all modals to the overlay element.
+   *
+   * While aria-owns is not required for the ARIA 1.2 `role="combobox"` interaction pattern,
+   * it fixes an issue with VoiceOver when the autocomplete appears inside of an `aria-model="true"`
+   * element (e.g. a dialog). Without this `aria-owns`, the `aria-modal` on a dialog prevents
+   * VoiceOver from "seeing" the autocomplete's listbox overlay for aria-activedescendant.
+   * Using `aria-owns` re-parents the autocomplete overlay so that it works again.
+   * See https://github.com/angular/components/issues/20694
+   */
+  private _exposeToModals() {
+    // TODO(http://github.com/angular/components/issues/26853): consider de-duplicating this with
+    // the `LiveAnnouncer` and any other usages.
+    const id = this.autocomplete.id;
```

Comment:
```suggestion
    const panelId = this.autocomplete.id;
```
This makes it clear _which_ `id` we're using here